frontend/src/routes/SignIn/index.js

```diff
@@ -44,10 +49,12 @@ export default function SignIn() {
     event.preventDefault();
     if (!formValues.email || formValues.email === "") {
       document.getElementById("incomplete_error").innerText =
-        "Please fill your email address";
+        "Please enter your email address";
     } else if (!formValues.password || formValues.password === "") {
       document.getElementById("incomplete_error").innerText =
-        "Please fill your password";
+        "Please enter your password";
+    } else if (!validateEmail(formValues.email)) {
+      document.getElementById("incomplete_error").innerText = "";
```

Comment:
There is a bug here. If the user clicks on next without entering the city, ` next_count` will be 1 now. Then, the user enter the city, but does not enter a valid email address. It shows a warning on the top. Then, the user enters both city and valid email, but the next button does not show, which makes the user unable to proceed to the next page. You should reset `next_count` to 0 whenever it gives the warning that is not "Please fill your city ...".

Reply:
should be fixed now